packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -753,3 +765,435 @@ Existing domains can be imported into CDK apps using `UserPoolDomain.fromDomainN
 ```ts
 const myUserPoolDomain = cognito.UserPoolDomain.fromDomainName(this, 'my-user-pool-domain', 'domain-name');
 ```
+
+## Identity Pools
+
+Identity pools provide temporary AWS credentials for users who are guests (unauthenticated) and for users who have been  
+authenticated and received a token. An identity pool is a store of user identity data specific to your account.
+
+Identity pools can be used in conjunction with Cognito User Pools or by accessing external federated identity providers  
+directly.
+
+A basic Identity Pool with minimal configuration has no required props, with default authenticated and unauthenticated  
+roles applied to the identity pool: 
+
+```ts
+new cognito.IdentityPool(this, 'myIdentityPool');
+```
+
+### Authenticated and Unauthenticated Identities
+
+Identity pools define two types of identities: authenticated and unauthenticated. Every identity in your identity pool  
+is either authenticated or unauthenticated. Authenticated identities belong to users who are authenticated by a public  
+login provider (Amazon Cognito user pools, Login with Amazon, Sign in with Apple, Facebook, Google, SAML, or any OpenID  
+Connect Providers) or a developer provider (your own backend authentication process). Unauthenticated identities  
+typically belong to guest users.
+
+You can add permissions to your default authenticated role to determine what base permissions an authenticated  
+user will have:
+
+```ts
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  userPermissions: [
+    {
+      effect: iam.Effect.ALLOW,
+      actions: ['execute-api:*'],
+      resources: ['*'],
+    },
+  ],
+});
+```
+
+If you want to allow guest users limited access, you can add permissions to the default unauthenticated role as well:
+
+```ts
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  userPermissions: [
+    {
+      effect: iam.Effect.ALLOW,
+      actions: ['dynamodb:*'],
+      resources: ['*'],
+    },
+  ],
+  guestPermissions: [
+    {
+      effect: iam.Effect.ALLOW,
+      actions: ['dynamodb:Get*'],
+      resources: ['*'],
+    },
+  ],
+});
+```
+
+The default authenticated and unauthenticated roles are preconfigured to allow users to assume their respective  
+credentials using the `sts:AssumeRoleWithWebIdentity` action. You can use a different action to assume the role with  
+the `assumeAction` field:
+
+```ts
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  assumeAction: 'sts:AssumeRole',
+  userPermissions: [
```

Comment:
I agree with your comments above and am going to update the added methods to follow the `grant` model, my only reservation with not also providing these options is that then with the `IdentityPool` minimal setup configuration, the default authenticated and default unauthenticated roles will in effect do the same thing. 

They would both allow their respective users to assume the role, but nothing more. This can work for the unauthenticated role because it should probably default to denying access, but the authenticated role should have some permissions attached. The authenticated role could have a default policy attached, but since different apps require permissions for different services for their users, I don't think we can predict what that policy would look like.

What do you think about replacing these fields with a `grantActions` type field that could add a policy to the default authenticated role upon creation?